File: Labs/C++/Lab_14/Lab_14.cpp
// Lab 14
// Derek P Sifford
#include <iostream>
using namespace std;

void change(int value, int &hundreds, int &fifties, int &twenties,
                int &tens, int &fives, int &ones) {

    if (value > 100) {
        hundreds = value / 100;
        value = value % 100;
    }

    if (value > 50) {
        fifties = value / 50;
        value = value % 50;
    }

    if (value > 20) {
        twenties = value / 20;
        value = value % 20;
    }

    if (value > 10) {
        tens = value / 10;
        value = value % 10;
    }

    if (value > 5) {
        fives = value / 5;
        value = value % 5;
    }

    if (value > 1) {
        ones = value / 1;
        value = value % 1;
    }

}


int main() {

    int value, hundreds = 0, fifties = 0, twenties = 0, tens = 0, fives = 0, ones = 0;

    cout << "Enter an integer value" << endl;
    cin >> value;

    change(value, hundreds, fifties, twenties, tens, fives, ones);

    printf(
        "Your change is:\n"
        "%d hundreds\n"
        "%d fifties\n"
        "%d twenties\n"
        "%d tens\n"
        "%d fives\n"
        "%d ones\n",
        hundreds, fifties, twenties, tens, fives, ones
    );
}

File: Labs/golang/Lab_14/Lab_14.go
package main

import (
	"fmt"
)

type currency struct {
	hundreds, fifties, twenties, tens, fives, ones int
}

func (c *currency) change(v int) {

	if v > 100 {
		c.hundreds = v / 100
		v = v % 100
	}
	if v > 50 {
		c.fifties = v / 50
		v = v % 50
	}
	if v > 20 {
		c.twenties = v / 20
		v = v % 20
	}
	if v > 10 {
		c.tens = v / 10
		v = v % 10
	}
	if v > 5 {
		c.fives = v / 5
		v = v % 5
	}
	if v > 1 {
		c.ones = v / 1
		v = v % 1
	}

}

func main() {

	var input int
	c := currency{}

	fmt.Println("Enter any integer")
	fmt.Scanln(&input)

	c.change(input)
	fmt.Printf(
		"Your change is:\n"+
			"%v hundreds\n"+
			"%v fifties\n"+
			"%v twenties\n"+
			"%v tens\n"+
			"%v fives\n"+
			"%v ones\n",
		c.hundreds, c.fifties, c.twenties, c.tens, c.fives, c.ones)

}
